### PR TITLE
Adding changes to Pedro Arduino use case

### PR DESCRIPTION
**Pull request by parduino**
## Overview

…

## Changes

…

## Testing

1.

## UI

…

### CODE REVIEW

**Comment by parduino** [comment]
Adding changes to Pedro Arduino use case

**Comment by parduino** [comment]
Please let me know if this creates problems.
I only changes Arduino files.

**Comment by silviamazzoni** [comment]
@parduino these are just editorial changes, correct?
They look good to me.
It does look like you did work with the latest version of the docs, so it looks great to me.
Anything I should look at specifically? otherwise I'll approve it.

**Comment by parduino** [comment]
Hi Silvia,

  Yes, mostly editorial changes and cleaning stuff.
Basic Markdown is very limited, so it has to be complemented with functionality either from HTML or CSS.
Unfortunately, the system used at DesignSafe does not allow much of this functionality.
So, I have been trying to use as much plain markdown as possible.

I will be testing all the notebooks next. They all point to /community in DesignSafe and should work, but Tapis apps constantly change (particularly in name) and python dependencies too. So, I will take a look at that too.

So, It is very possible I will have another pull request…

Pedro

**Comment by silviamazzoni** [comment]
@wesleyboar i'm happy to have this update merged.
@parduino yes MD is limited, but Wesley and I have been working on adding functionality.
In working with Wesley, I learned that, afterall, md is just a shortcut to html, a way to simplify the text and keep a uniform style. so we should minimize raw html and css code. Wes can create "any" md cod you'd like. so he and I will have a look at your edits and see what Wesley can do to make it a useable style.
we need consistency across all use-case documentation.
thank you for this.

**Comment by silviamazzoni** [comment]
@parduino i see you reduced the figure sizes.... i'm afraid this may lose resolution.
Was the reduction due to display issues? @wesleyboar can you do your magic on that? so that we don't have to reduce image quality?

**Comment by parduino** [comment]
Hi Silvia,

    Yes, one of the problems using plain markdown is that you can’t control sizes/scaling/centering of figures and tables.
This is why I usually used HTML commands in markdown files.
It works beautifully on my end but has to work on the end where these things are posted.

So, yes, I changed the size of all figures trying to fit sizes I think can be used here. But, this is on my monitor…

I wonder how all this looks on your end.

If possible I would recommend to add functionality for:


a)       Centering and scaling of figures and tables

b)      Basic equations functionality


If it does not look good, please don’t accept the pull request yet. I can make changes to make this look well.

Thanks!!

PEdro

**Comment by parduino** [comment]
Hi Silvia,

  Wait, don’t accept the pull request. I see the images loose quality when I try to resize them to fit markdown…
Working on this.
Thanks

PEdro

File: user-guide/docs/usecases/arduino/usecase.md [hunk 1]
```diff
@@ -17,12 +17,15 @@ A collection of educational notebooks to introduce model-parameter calibration a
...
 #### Jupyter Notebooks
-The following Jupyter notebooks are available to facilitate the analysis of each case. They are described in details in this section. You can access and run them directly on DesignSafe by clicking on the "Open in DesignSafe" button.
+The following Jupyter notebooks are made available to facilitate the analysis of each case. They are described in detail in this section. You can access and run them directly on DesignSafe by clicking on the "Open in DesignSafe" button.
 
 | Site Response | Notebook |
 | :-------: | :---------:  |
 | FreeField Response | freeFieldJupyterPM4Sand_Community.ipynb <br> [![Open In DesignSafe](https://raw.githubusercontent.com/geoelements/LearnMPM/main/DesignSafe-Badge.svg)](https://jupyter.designsafe-ci.org/hub/user-redirect/lab/tree/CommunityData/Jupyter%20Notebooks%20for%20Civil%20Engineering%20Courses/University_of_Washington/freeFieldJupyterPM4Sand/freeFieldJupyterPM4Sand_Community.ipynb) |
 
+---
+---
```
Comment (wesleyboar) [inline user-guide/docs/usecases/arduino/usecase.md]: If you are trying to draw a [horizontal rule](https://www.markdownguide.org/basic-syntax/#horizontal-rules), you may use just one `---` surrounded by new lines.

```suggestion
---
```

<details><summary>What does two consecutive `---` lines render?</summary>

An `<h2>---</h2>` i.e.:

## ---

</details>

File: user-guide/docs/usecases/arduino/usecase_quoFEM.md [hunk 2]
```diff
@@ -50,10 +49,9 @@ In this context, the notebooks included in this use-case complement input genera
 
 To connect SimCenter applications and Jupyter notebooks in DesignSafe, it's essential to ensure that all required tools are accessible from both frameworks. The SimCenter's vision is well-aligned with this concept and offers all of the necessary functionality through backend applications installed in DesignSafe that can be accessed via Tapis apps. This is schematically shown in Fig 2. Additionally, all SimCenter workflows are stored in JSON files that represent all steps in a workflow. This file is readily accesible using a JSON parser.
 
-<p align="center">
-<img src="../img/UC2-Arduino-2.png" alt="quoFEm in DesignSafe" width="600"/>
-</p>
-<p align="center"> <b>Fig.2 - Running quoFEM analysis using remote computing resources at DesignSafe</b> </p>
+![quoFEm in DesignSafe](img/UC2-Arduino-2.png "Probabilistic calibration")
+<p style="quoFEm in DesignSafe"> Fig.2 - Running quoFEM analysis using remote computing resources at DesignSafe </p>
```
Comment (wesleyboar) [inline user-guide/docs/usecases/arduino/usecase_quoFEM.md]: 🙂

```suggestion
<p style="text-align: center;"> Fig.2 - Running quoFEM analysis using remote computing resources at DesignSafe </p>
```

**Review by wesleyboar — COMMENTED** [review]
### Concerns

1. The [downsized] images, because some [have text] that is now not legible.
    <sup>Are you comfortable undoing those or explaining the intent?</sup>
2. [The double `---` I mention.](https://github.com/DesignSafe-CI/DS-User-Guide/pull/47#discussion_r1644955322)

### Answers

a) Centering and scaling
- **figures**: You may continue to use `<p style="text-align: center;"> Fig.1 - Schematic of Matlab workflow</p>` for now. I expect it. Solution coming.[^1]
- **tables**:  Will do, specific to DS-User-Guide. Please open a GitHub issue to describe what scaling you expect. (Centering, I grok.)

b) basic equations functionality

- Reported ([#44](https://github.com/DesignSafe-CI/DS-User-Guide/issues/44)). Agreed. Coming.

[^1]: Using `<p style="align-text="center">` with `[img](path)` is cleaner and on my list of known implementations to update or support when I switch to new solution. I want us to use [`<figure>` + `<figcaption>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figure). Usage could be (a) 😐 directly, [like TACC](https://github.com/TACC/TACC-Docs/blob/v0.6.0/docs/basics/datatransfer.md?plain=1#L39-L40); (b) 🙁 indirectly, with [`/// html | figure` + `/// html | figcaption`](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/html/#usage); (c) 🙂 automatically, with [`mkdocs-caption`](https://github.com/tobiasah/mkdocs-caption#readme).

File: user-guide/docs/usecases/arduino/usecase_quoFEM.md (hunk 2 above)
Comment (parduino) [reply user-guide/docs/usecases/arduino/usecase_quoFEM.md]: Hi Wesley and Silvia,

  Let me play a bit with images... Yes, I made them smaller and by doing that may have affected resolution. 
  Working on them.  Don't kill the pull request. I will make changes and add them to the same pull.
  Thanks

Comment (wesleyboar) [reply user-guide/docs/usecases/arduino/usecase_quoFEM.md]: Okay. Here's a preview of your work as of 4743e8d:
https://pprd.docs.tacc.utexas.edu/usecases/arduino/usecase_quoFEM/
<sup>Disregard the domain. It's the only server to which I can ad-hoc deploy docs.</sup>

Also, I am available to help you if you have trouble [Testing locally](https://github.com/DesignSafe-CI/DS-User-Guide?tab=readme-ov-file#testing).

**Comment by parduino** [comment]
Hi Wesley and Silvia,

How does it look right now??
I made further changes and brought back images with the original size

**Comment by silviamazzoni** [comment]
I'll leave it to @wesleyboar at this point, he has the setup to see how it looks.

File: user-guide/docs/usecases/arduino/usecase.md (hunk 1 above)
Comment (wesleyboar) [reply user-guide/docs/usecases/arduino/usecase.md]: As of a744abd, this still produces giant "---":
a744abd double hrs
(https://pprd.docs.tacc.utexas.edu/usecases/arduino/usecase/)

**Comment by wesleyboar** [comment]
Okay. Here's a preview of the changes as of a744abd:
https://pprd.docs.tacc.utexas.edu/usecases/arduino/usecase_quoFEM/
<sup>Disregard the domain. It's the only server to which I can ad-hoc deploy docs.</sup>

I am unavailable to eagle-eye review until Monday.

I still suggest the change I proposed at https://github.com/DesignSafe-CI/DS-User-Guide/pull/47#discussion_r1644955322.

**Comment by wesleyboar** [comment]
_P.S. Everyone can have "the setup". There are instructions for [testing locally](https://github.com/DesignSafe-CI/DS-User-Guide?tab=readme-ov-file#testing)._

File: user-guide/docs/usecases/arduino/FreeField-JupyterNB/N10_T3.tcl
Comment (wesleyboar) [inline user-guide/docs/usecases/arduino/FreeField-JupyterNB/N10_T3.tcl:1]: File still referenced in doc [`arduino/usecase_siteResponse.md` line 28](https://github.com/DesignSafe-CI/DS-User-Guide/blob/pr/47/user-guide/docs/usecases/arduino/usecase_siteResponse.md?plain=1#L28) ([live page](https://www.designsafe-ci.org/user-guide/usecases/arduino/usecase_siteResponse/)) which is referenced by [`arduino/usecase.md` line 71](https://github.com/DesignSafe-CI/DS-User-Guide/blob/pr/47/user-guide/docs/usecases/arduino/usecase.md?plain=1#L71) ([live page](https://www.designsafe-ci.org/user-guide/usecases/arduino/usecase_siteResponse/)).

<details><summary>Screenshots</summary>

| file reference | link to doc that ref's file |
| - | - |
| usecase_siteResponse | usecase |

</details> 

❓ Remove reference? Retain file? Resolve later?

File: user-guide/docs/usecases/arduino/FreeField-JupyterNB/interactiveplot.py
Comment (wesleyboar) [inline user-guide/docs/usecases/arduino/FreeField-JupyterNB/interactiveplot.py:1]: [File still referenced in docs.](https://github.com/DesignSafe-CI/DS-User-Guide/blob/pr/47/user-guide/docs/usecases/arduino/usecase_siteResponse.md#L226)

❓ Remove reference? Retain file? Resolve later?

File: user-guide/docs/usecases/arduino/FreeField-JupyterNB/ShortReport.rst
Comment (wesleyboar) [inline user-guide/docs/usecases/arduino/FreeField-JupyterNB/ShortReport.rst:1]: File still referenced in docs:
- [`arduino/usecase_matlab.md`](https://github.com/DesignSafe-CI/DS-User-Guide/blob/pr/47/user-guide/docs/usecases/arduino/usecase_matlab.md?plain=1#L248)
- [`arduino/usecase_siteResponse.md`](https://github.com/DesignSafe-CI/DS-User-Guide/blob/pr/47/user-guide/docs/usecases/arduino/usecase_siteResponse.md#L192)

❓ Remove references? Retain files? Resolve later?

File: user-guide/docs/usecases/arduino/FreeField-JupyterNB/schematic.png
Comment (wesleyboar) [inline user-guide/docs/usecases/arduino/FreeField-JupyterNB/schematic.png:1]: [File still referenced in docs.](https://github.com/DesignSafe-CI/DS-User-Guide/blob/pr/47/user-guide/docs/usecases/arduino/usecase_siteResponse.md?plain=1#L30)

❓ Remove reference? Retain file? Resolve later?

File: user-guide/docs/usecases/arduino/FreeField-JupyterNB/velocity.input
Comment (wesleyboar) [inline user-guide/docs/usecases/arduino/FreeField-JupyterNB/velocity.input:1]: [File still referenced in docs.](https://github.com/DesignSafe-CI/DS-User-Guide/blob/pr/47/user-guide/docs/usecases/arduino/usecase_siteResponse.md?plain=1#L28)

❓ Remove reference? Retain file? Resolve later?

**Review by wesleyboar — COMMENTED** [review]
Some files are deleted that user guide still references. How do you want to handle broken links… Remove references? Retain files? Resolve later?

File: user-guide/docs/usecases/arduino/FreeField-JupyterNB/velocity.input
Comment (parduino) [reply user-guide/docs/usecases/arduino/FreeField-JupyterNB/velocity.input:1]: Hi Wesley,

Thanks for chcecking this. You are correct, by erasing a folder some links were lost.
To fix this I added the referenced files in a "files" folder.
My idea was to put all these files in community where the notebooks are, but, this seems more efficient.

Please let me know if you see any other problem.

Next I will check the notebooks in community... They tend to become deprecated when tapis apps are modified...

**Review by wesleyboar — APPROVED** [review]
The latest commits should have fixed the broken links. Approved.

**Comment by wesleyboar** [comment]
Merging based on my own approval and https://github.com/DesignSafe-CI/DS-User-Guide/pull/47#issuecomment-2177323389.

**Pull request merged**